sources/logger.c

```diff
@@ -484,3 +484,49 @@ void od_logger_write(od_logger_t *logger, od_logger_level_t level,
 		_od_logger_write(logger, output, len, level);
 	}
 }
+
+// TODO: decide on how to pass and write context info
+extern void od_logger_write_plain(od_logger_t *logger, od_logger_level_t level,
+				  char *context, void *client, void *server,
+				  char *string)
+{
+	if (logger->fd == -1 && !logger->log_stdout && !logger->log_syslog)
+		return;
+
+	if (level == OD_DEBUG) {
+		int is_debug = logger->log_debug;
+		if (!is_debug) {
+			od_client_t *client_ref = client;
+			od_server_t *server_ref = server;
+			if (client_ref && client_ref->rule) {
+				is_debug = client_ref->rule->log_debug;
+			} else if (server_ref && server_ref->route) {
+				od_route_t *route = server_ref->route;
+				is_debug = route->rule->log_debug;
+			}
+		}
+		if (!is_debug)
+			return;
+	}
+
+	int len = strlen(string);
+	char output[len + OD_LOGLINE_MAXLEN];
```

Comment:
why not `char output[len + 1];` ?

Reply:
Because all info in logger.format must fit in there.